microsoft/SqlNexus
# Fixing the PowerBI Performance Report

Changed a script that was having issues when collecting perfmon data from FCI instances. To get the machine name and find the counter, instead of using SERVERPROPERTY('ComputerNamePhysicalNetBIOS'), it was using SERVERPROPERTY('MachineName'), which returns the SQL virtual name. The perfmon counters have the physical machine name always, even if this is a FCI.

Also changed a script that was causing issues when the instance where the nexus database is created doesn't have the collation  SQL_Latin1_General_CP1_CI_AS. Now it should work with any collation

How to test:

1 - Perfmon data issue with FCI instances

- Open nexus

- Open the PowerBI Performance Report

- Connect to a nexus database that has data from a FCI instance (before the fix, multiple charts don't open with an error about an inexistent counter)

2 - Collation Issue

In case you have a nexus database created on a SQL instance that doesn't have the collation SQL_Latin1_General_CP1_CI_AS, run the PowerBI Performance Report against it. In the current version, it will fail with a conflict collation error when running the DatabaseFiles report. With this version, it should work with any instance collation
